form/src/main/java/org/dhis2/form/ui/provider/inputfield/SignatureProvider.kt

```diff
@@ -27,24 +34,42 @@ fun ProvideInputSignature(
             ?.asImageBitmap()
     }
 
+    var uploadState by remember(fieldUiModel) { mutableStateOf(getUploadState(fieldUiModel.displayName, fieldUiModel.isLoadingData)) }
+
     InputSignature(
         modifier = modifier,
         title = fieldUiModel.label,
         state = fieldUiModel.inputState(),
         supportingText = fieldUiModel.supportingText(),
         legendData = fieldUiModel.legend(),
         isRequired = fieldUiModel.mandatory,
-        uploadState = if (fieldUiModel.value != null) UploadState.LOADED else UploadState.ADD,
+        uploadState = uploadState,
         load = { imageBitmap },
         painterFor = imageBitmap?.let {
             {
                     image ->
                 BitmapPainter(image!!)
             }
         },
-        onDownloadButtonClick = { fieldUiModel.invokeUiEvent(UiEventType.SHOW_PICTURE) },
-        onResetButtonClicked = { fieldUiModel.onClear() },
-        onAddButtonClicked = { fieldUiModel.invokeUiEvent(UiEventType.ADD_SIGNATURE) },
-        onImageClick = { fieldUiModel.invokeUiEvent(UiEventType.SHOW_PICTURE) },
+        onDownloadButtonClick = {
+            uiEventHandler.invoke(RecyclerViewUiEvents.OpenFile(fieldUiModel))
+        },
+        onResetButtonClicked = {
+            fieldUiModel.onClear()
+            uploadState = getUploadState(fieldUiModel.displayName, false)
+            intentHandler.invoke(
+                FormIntent.OnAddImageFinished(
+                    uid = fieldUiModel.uid,
+                ),
+            )
+            fieldUiModel.onClear()
```

Comment:
There are two calls to onClear on this lambda function

Reply:
nice catch! I'll update it thanks